dbuf-lsp/src/core/dbuf_language.rs

```diff
@@ -0,0 +1,60 @@
+//! Contains dbuf language specific information, like builtin types or constants.
+//!
+
+use std::{collections::HashSet, sync::OnceLock};
+
+static BUILDIN_TYPES: OnceLock<HashSet<String>> = OnceLock::new();
+static KEYWORDS: OnceLock<HashSet<String>> = OnceLock::new();
+
+/// Returns buildint types set.
+pub fn get_bultin_types() -> &'static HashSet<String> {
+    BUILDIN_TYPES.get_or_init(|| {
+        let mut m = HashSet::new();
+        let types = ["Int", "String", "Bool", "Unsigned", "Float"];
+        types.iter().for_each(|&s| {
+            m.insert(s.to_string());
+        });
+        m
+    })
+}
+
+/// Returns dbuf keywords set.
+pub fn get_keywords() -> &'static HashSet<String> {
+    KEYWORDS.get_or_init(|| {
+        let mut m = HashSet::new();
+        let types = ["message", "enum"];
+        types.iter().for_each(|&s| {
+            m.insert(s.to_string());
+        });
+        m
+    })
+}
+
+/// Checks if `type_name` is correct name for Type or Constructor.
+pub fn is_correct_type_name(type_name: &str) -> bool {
```

Comment:
Instead of checking _if_ something _is_ something-else, I'd introduce a separate type for this something-else
In this module, the types `TypeName` and `FieldName` with corresponding `TryFrom<&str>` instances
It's particular manifestation of a [newtype pattern](https://www.lurklurk.org/effective-rust/newtype.html) where we carry information about the invariant in the type

Reply:
Also see https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/